src/gui/guiutility.cpp

```diff
@@ -481,28 +481,20 @@ qint64 GuiUtility::folderDiskSize(const QString &dirPath) {
 }
 
 bool GuiUtility::openFolder(const QString &dirPath) {
-    if (!dirPath.isEmpty()) {
-        QFileInfo fileInfo(dirPath);
-        if (fileInfo.exists()) {
-            const QUrl url = getUrlFromLocalPath(fileInfo.path());
-            if (url.isValid()) {
-                if (!QDesktopServices::openUrl(url)) {
-                    return false;
-                }
-            }
-        } else if (fileInfo.dir().exists()) {
-            const QUrl url = getUrlFromLocalPath(fileInfo.dir().path());
-            if (url.isValid()) {
-                if (!QDesktopServices::openUrl(url)) {
-                    return false;
-                }
-            } else {
-                return false;
-            }
-        } else {
-            return false;
-        }
+    if (dirPath.isEmpty()) return true;
+
+    QFileInfo fileInfo(dirPath);
+    if (fileInfo.exists()) {
+        const QUrl url = getUrlFromLocalPath(QDir::cleanPath(fileInfo.filePath()));
+        if (url.isValid() && !QDesktopServices::openUrl(url)) return false;
+    } else if (fileInfo.dir().exists()) {
```

Comment:
@ClementKunz  Here is the fallback strategy followed when the path indicates an invalid item: we try attempt to open the parent folder, if any.